### PR TITLE
basic DFA support

### PR DESCRIPTION
**Pull request by cjeffery**

### CODE REVIEW

**Review by Jafaral — CHANGES_REQUESTED** [review]
Looks good overall. I added comment regarding the use of existing top level Makedefs and Makedef.uni. In this case we may only need to include Makedefs.uni but we need to add CC to it to get the configured C compiler instead of including the two Makedef files and I'm not sure if there are collisions between the two.

File: uni/uflex/Makefile [hunk 1]
```diff
@@ -6,16 +6,15 @@
 # Based on Katie Ray's Project: ulex, March 15, 2003. Katie's unfinished
 # C program is preserved in the deprecated legacy directory ulex/
 #
-CC=cc
+CC=gcc
```
[the remark this answers is not in the document]
Comment (cjeffery) [reply uni/uflex/Makefile:9]: I agree with this change.  uflex has frequently been built in standalone directories outside of unicon/ but within the Unicon distribution we should make it as clean and portable as we can.

File: uni/uflex/Makefile [hunk 2]
```diff
@@ -32,12 +31,24 @@ uflex-lex.u: uflex-lex.icn
...
+
 tree.u: tree.icn
 	unicon -c tree
 
+nfa2dfa: nfa2dfa.icn
+	unicon -o nfa2dfa nfa2dfa.icn
+
+dfa2auto: dfa2auto.icn 
+	unicon -o dfa2auto dfa2auto.icn	ulexskel
+
```
[the remark this answers is not in the document]
Comment (cjeffery) [reply uni/uflex/Makefile:51]: Sounds good.

File: uni/uflex/Makefile [hunk 3]
```diff
@@ -6,16 +6,15 @@
 # Based on Katie Ray's Project: ulex, March 15, 2003. Katie's unfinished
 # C program is preserved in the deprecated legacy directory ulex/
 #
-CC=cc
+CC=gcc
 CFLAGS= -c -g
-LEX=flex
-YACC=yacc
+LEX=win_flex
+YACC=win_bison -y
```
[the remark this answers is not in the document]
Comment (cjeffery) [reply uni/uflex/Makefile:12]: I think these are vestigial references to flex and bison used for building C ulex and not necessary here.

**Pull request merged**